include/triton/Analysis/Allocation.h

```diff
@@ -18,16 +18,39 @@ namespace mlir {
 namespace triton {
 class AllocationAnalysis;
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec);
-SmallVector<unsigned> getScratchConfigForCvtLayout(RankedTensorType srcType,
-                                                   RankedTensorType dstType,
-                                                   unsigned &inVec,
-                                                   unsigned &outVec);
-SmallVector<unsigned> getRepShapeForCvtLayout(triton::gpu::ConvertLayoutOp op);
-SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
-                                              RankedTensorType dstTy);
+struct ScratchConfig {
+  SmallVector<unsigned> repShape;
+  SmallVector<unsigned> paddedRepShape;
+  unsigned inVec{1};
+  unsigned outVec{1};
+  unsigned paddedDim;
+  unsigned paddedSize{0};
+  unsigned paddedStride;
+
+  ScratchConfig() = default;
```

Comment:
I don't understand why we have defaults for some member variables but not others.  Won't that mean that when you call the default constructor, some of the variables are left uninitialized?  That seems undesirable to me.

I only see one place where the default constructor is called, and we're still building a non-default ScratchConfig there.  In that case, perhaps just remove the default constructor?